### PR TITLE
Assign a structure for KeyIsLocked content

### PR DESCRIPTION
**Pull request by breeswish**
Signed-off-by: Breezewish <breezewish@pingcap.com>

## What have you changed? (mandatory)

This PR assigns a structure (i.e. `LockInfo`) for holding KeyIsLocked content, so that passing these content do not need to repeatedly write out fields.

As you can see, this change will eliminate total LOC.

## What are the type of the changes? (mandatory)

- Engineering (engineering change which doesn't change any feature or fix any issue)

## How has this PR been tested? (mandatory)

Existing tests.

### CODE REVIEW

File: src/storage/lock_manager/util.rs [hunk 1]
```diff
@@ -52,18 +52,14 @@ pub fn extract_raw_key_from_process_result(pr: &ProcessResult) -> &[u8] {
...
         let key = Key::from_raw(&raw_key);
         let ts = 100;
-        let case = StorageError::from(TxnError::from(MvccError::KeyIsLocked {
-            key: raw_key,
-            primary: vec![],
-            ts,
-            ttl: 100,
-            txn_size: 0,
-        }));
+        let case = StorageError::from(TxnError::from(MvccError::KeyIsLocked(LockInfo::default())));
```
Comment (youjiali1995) [inline src/storage/lock_manager/util.rs]: Forget to set `LockInfo`.

File: src/storage/mvcc/txn.rs [hunk 2]
```diff
@@ -225,13 +225,13 @@ impl<S: Snapshot> MvccTxn<S> {
...
-                    ts: lock.ts,
-                    ttl: lock.ttl,
-                    txn_size: options.txn_size,
-                });
+                let mut info = kvproto::kvrpcpb::LockInfo::default();
+                info.set_primary_lock(lock.primary);
+                info.set_lock_version(lock.ts);
+                info.set_key(key.into_raw()?);
+                info.set_lock_ttl(lock.ttl);
+                info.set_txn_size(options.txn_size);
```
Comment (MyonKeminta) [inline src/storage/mvcc/txn.rs]: Is it better if we can put these things in a function?

Comment (breeswish) [reply src/storage/mvcc/txn.rs]: Function arguments are unnamed, may make it harder to read?

File: src/storage/lock_manager/util.rs [hunk 3]
```diff
@@ -52,18 +52,18 @@ pub fn extract_raw_key_from_process_result(pr: &ProcessResult) -> &[u8] {
...
-            key: raw_key,
-            primary: vec![],
-            ts,
-            ttl: 100,
-            txn_size: 0,
-        }));
+        let mut info = LockInfo::default();
+        info.set_key(raw_key);
+        info.set_lock_version(ts);
+        info.set_ttl(100);
```
Comment (youjiali1995) [inline src/storage/lock_manager/util.rs]: s/set_ttl/set_lock_ttl

File: src/storage/mvcc/txn.rs (hunk 2 above)
Comment (MyonKeminta) [reply src/storage/mvcc/txn.rs]: Oh, that's reasonable.

**Pull request merged**